
syntax = "proto3";

package RegistryServer;


message ServerRequest {
	string name = 1;
	string address = 2;	
}

message ServerResponse {
	string name = 1;
	string address = 2;
}



message ClientRequest {
	string client_id = 1;
}

message ServerList {
	repeated ServerRequest servers = 1;
}

message ClientResponse {
	ServerList serverList = 1;
}



message PrimaryResponse {
	string response = 1;
}



service RegisterService {
	rpc Register (ServerRequest) returns (ServerResponse);
}

service GetServerListService {
	rpc GetServerList (ClientRequest) returns (ClientResponse);
}

service InformPrimaryServerService {
	rpc InformPrimaryServer (ServerRequest) returns (PrimaryResponse);
}
